src/ripple/shamap/SHAMap.h

```diff
@@ -358,66 +362,109 @@ SHAMap::setUnbacked ()
 
 //------------------------------------------------------------------------------
 
-class SHAMap::iterator
-    : public boost::iterator_facade<
-        SHAMap::iterator,
-            std::shared_ptr<SHAMapItem const> const,
-                std::forward_iterator_tag>
+class SHAMap::const_iterator
```

Comment:
I'm wondering if you can get rid of the `#include <.../iterator_facade.hpp>` now?